lexxpluss_apps/src/zcan_uss.hpp

```diff
@@ -37,32 +37,30 @@ namespace lexxhard {
 class ros_uss {
 public:
     void init() {
-        /* TODO 初期化が他に必要か調べる */
-        dev = device_get_binding("CAN_2");  /* TODO CAN_2 かどうか調べる（デバイス定義） */
+        dev = device_get_binding("CAN_2"); 
     }
     void poll() {
         uss_controller::msg message;
 
         while (k_msgq_get(&uss_controller::msgq, &message, K_NO_WAIT) == 0) {
-            uint8_t packedData[CAN_DATA_LENGTH_USS] {0}; // 8バイトのデータ配列 0で初期化
+            uint8_t packedData[CAN_DATA_LENGTH_USS] {0}; 
+            uint16_t data1 = (uint16_t)(message.front_left / 2);
+            uint16_t data2 = (uint16_t)(message.front_right / 2);
+            uint16_t data3 = (uint16_t)(message.left / 2);
+            uint16_t data4 = (uint16_t)(message.right / 2);
+            uint16_t data5 = (uint16_t)(message.back / 2);
 
-            /* TODO USSのデータ出力サイズを調べる(2mm解像度で12bit?) */
-            uint16_t data1 = (uint16_t)(message.front_left / 1000);
-            uint16_t data2 = (uint16_t)(message.front_right / 1000);
-            uint16_t data3 = (uint16_t)(message.left / 1000);
-            uint16_t data4 = (uint16_t)(message.right / 1000);
-            uint16_t data5 = (uint16_t)(message.back / 1000);
+            // 8-byte : | byte0 | byte1 | byte2 | byte3 | byte4 | byte5 | byte6 | byte7 |
+            // 8-byte : |   data1   |   data2   |   data3   |   data4   |   data5   | x |
+            packedData[0] = (data1 & 0xFF0) >> 4;                         
+            packedData[1] = ((data1 & 0x00F) << 4) | (data2 >> 8);   
```

Comment:
```suggestion
            packedData[1] = ((data1 & 0x00F) << 4) | ((data2 & 0xF00) >> 8);   
```

現状の実装では、data2の上位4ビットに何かデータがあった場合、意図しない値になる。
心配しすぎかもしれません。
With the current implementation, if there is any data in the upper 4 bits of data2, it will be an unintended value.
I may be worrying too much.